### ISSUE TITLE
The epoch process is suspended without reporting an error or continuing to execute

### ISSUE THREAD

**Issue by love381** [feferna/psoCNN#5]
Stuck 
Initialization - Particle: 4
conv | fc | fc | fc | 
Epoch 1/1

   32/60000 [..............................] - ETA: 47:00 - loss: 7.0256 - accuracy: 0.0938
   64/60000 [..............................] - ETA: 25:55 - loss: 6.9689 - accuracy: 0.1094

**Comment by feferna** [comment]
Are running tensorflow on GPU?

From the ETA, it looks like it is running on CPU.

**Comment by love381** [comment]
Yes i run it on GPU 960M------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;&quot;feferna&quot;<notifications@github.com&gt;
发送时间:&nbsp;2020年9月22日(星期二) 凌晨4:54
收件人:&nbsp;&quot;feferna/psoCNN&quot;<psoCNN@noreply.github.com&gt;;
抄送:&nbsp;&quot;love381&quot;<919485656@qq.com&gt;;&quot;Author&quot;<author@noreply.github.com&gt;;
主题:&nbsp;Re: [feferna/psoCNN] The epoch process is suspended without reporting an error or continuing to execute (#5)

**Comment by love381** [comment]
This is my GPU version. The framework requirements for the project installation are also the same, and there will still be a stuck phenomenon. The memory of the terminal GPU is not full. This project must use a server? ? ?
pciBusID: 0000:02:00.0 name: GeForce GTX 960M computeCapability: 5.0

**Comment by feferna** [comment]
It should work with any GPU, but, when Tensorflow fails to load the GPU libraries, it will run on CPU. An ETA of 47 minutes means that the code is running on CPU. Try to see Tensorflow warning messages at the beginning of the code. It should say that every library was loaded successfully.

**Comment by love381** [comment]
Using TensorFlow backend.
2020-09-22 18:55:43.015568: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_101.dll
2020-09-22 18:56:02.850400: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library nvcuda.dll
2020-09-22 18:56:03.625555: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 0 with properties: 
pciBusID: 0000:02:00.0 name: GeForce GTX 960M computeCapability: 5.0
coreClock: 1.176GHz coreCount: 5 deviceMemorySize: 4.00GiB deviceMemoryBandwidth: 74.65GiB/s
2020-09-22 18:56:03.626203: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_101.dll
2020-09-22 18:56:03.640168: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cublas64_10.dll
2020-09-22 18:56:03.854421: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cufft64_10.dll
2020-09-22 18:56:03.932132: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library curand64_10.dll
2020-09-22 18:56:04.085166: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusolver64_10.dll
2020-09-22 18:56:04.172760: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusparse64_10.dll
2020-09-22 18:56:04.221355: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll
2020-09-22 18:56:04.397375: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1703] Adding visible gpu devices: 0
2020-09-22 18:56:04.399168: I tensorflow/core/platform/cpu_feature_guard.cc:143] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2
2020-09-22 18:56:04.410751: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x1febee129d0 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2020-09-22 18:56:04.411526: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
2020-09-22 18:56:04.412622: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 0 with properties: 
pciBusID: 0000:02:00.0 name: GeForce GTX 960M computeCapability: 5.0
coreClock: 1.176GHz coreCount: 5 deviceMemorySize: 4.00GiB deviceMemoryBandwidth: 74.65GiB/s
2020-09-22 18:56:04.413222: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_101.dll
2020-09-22 18:56:04.414146: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cublas64_10.dll
2020-09-22 18:56:04.414563: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cufft64_10.dll
2020-09-22 18:56:04.415066: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library curand64_10.dll
2020-09-22 18:56:04.415591: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusolver64_10.dll
2020-09-22 18:56:04.416003: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusparse64_10.dll
2020-09-22 18:56:04.416381: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll
2020-09-22 18:56:04.417918: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1703] Adding visible gpu devices: 0
2020-09-22 18:56:15.642552: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1102] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-09-22 18:56:15.644067: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1108]      0 
2020-09-22 18:56:15.644849: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1121] 0:   N 
2020-09-22 18:56:15.648100: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1247] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 3032 MB memory) -> physical GPU (device: 0, name: GeForce GTX 960M, pci bus id: 0000:02:00.0, compute capability: 5.0)
2020-09-22 18:56:15.660905: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x1feeb221ac0 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:
2020-09-22 18:56:15.662304: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): GeForce GTX 960M, Compute Capability 5.0
Epoch 1/1
2020-09-22 18:56:17.656889: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cublas64_10.dll
2020-09-22 18:56:17.987863: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll
2020-09-22 18:56:19.153131: W tensorflow/stream_executor/gpu/redzone_allocator.cc:314] Internal: Invoking GPU asm compilation is supported on Cuda non-Windows platforms only
Relying on driver to perform ptx compilation. 
Modify $PATH to customize ptxas location.
This message will be only logged once.
----------------------------------------------------------------------------------------------------------------------------------------------
This is a warning issued by Tesnsorflow when my project started. According to the information on the Internet, no warning is fatal. But every time it reaches the fourth or fifth particle, it gets stuck. Can you help me see what's wrong with the warning? Thank you

**Comment by feferna** [comment]
From the warnings, it is indeed using your GPU. Thus, I have no idea why it gets stuck. Try to run with a smaller dataset such as the Convex one. It has a lot less training samples.

**Comment by love381** [comment]
Thank you very much.I will use else samples.Thank you,I will give your article a star.

**Comment by feferna** [comment]
Also, it is possible that this particle has a really big convolutional or fully-connected layer. You could try to reduce the number of output channels in the main.py ("max_conv_output_channels = 256") or the number of neurons in the fully-connected layers ("max_fully_connected_neurons = 300").

**Comment by love381** [comment]
Now I am trying a smaller data set, if it still fails, I will try to modify max_conv_output_channels and max_fully_connected_neurons. Thank you very much, your reply is very helpful to me.
